crates/biome_configuration/src/editorconfig.rs

```diff
@@ -0,0 +1,271 @@
+//! Helpers for handling .editorconfig files.
+//!
+//! Here's how the options are mapped to the formatter config:
+//!
+//! | .editorconfig option | biome option |
+//! |----------------------|--------------|
+//! | indent_style         | indent_style |
+//! | indent_size          | indent_width |
+//! | end_of_line          | line_ending  |
+//! | max_line_length      | line_width   |
+
+use std::{collections::HashMap, str::FromStr};
+
+use biome_deserialize::StringSet;
+use biome_diagnostics::Diagnostic;
+use biome_formatter::{LineEnding, LineWidth};
+use indexmap::IndexSet;
+use serde::{Deserialize, Deserializer};
+
+use crate::{
+    OverrideFormatterConfiguration, OverridePattern, Overrides, PartialConfiguration,
+    PartialFormatterConfiguration, PlainIndentStyle,
+};
+
+pub fn parse_str(s: &str) -> serde_ini::de::Result<EditorConfig> {
+    // TODO: use serde_path_to_error to emit better parse diagnostics
+    serde_ini::from_str(s)
+}
+
+/// Represents a parsed .editorconfig file, containing only the options that are relevant to biome.
+#[derive(Debug, Clone, Deserialize, Default)]
+#[serde(default)]
+pub struct EditorConfig {
+    #[serde(deserialize_with = "deserialize_bool_from_string")]
+    root: bool,
+    #[serde(flatten)]
+    options: HashMap<String, EditorConfigOptions>,
+}
+
+impl EditorConfig {
+    pub fn to_biome(
+        mut self,
+    ) -> (
+        Option<PartialConfiguration>,
+        Vec<EditorConfigValidationError>,
+    ) {
+        let diagnostics = self.validate();
+
+        let mut config = PartialConfiguration {
+            formatter: self.options.remove("*").map(|o| o.to_biome()),
+            ..Default::default()
+        };
+        let overrides: Vec<_> = self
+            .options
+            .into_iter()
+            .map(|(k, v)| OverridePattern {
+                include: Some(StringSet::new(IndexSet::from([k]))),
```

Comment:
Taking a look at the [spec of editorconfig](https://editorconfig.org/#file-format-details), I noted that Biome doesn't support all syntaxes of their glob patterns. Currently, Biome don't handle alternation like `{a,b}`. We should emit a warning if we encounter a pattern that includes a curly brace `{`.